velox/docs/functions/spark/math.rst

```diff
@@ -139,6 +139,11 @@ Mathematical Functions
         SELECT CAST(1 as DECIMAL(20, 3)) * CAST(0 as DECIMAL(20, 3)); -- decimal 0.000000
         SELECT CAST(201e-38 as DECIMAL(38, 38)) * CAST(301e-38 as DECIMAL(38, 38)); -- decimal 0.0000000000000000000000000000000000000
 
+.. spark:function:: nanvl(x, y) -> [same as x]
+
+    Returns ``x`` if it's not NaN, or ``y`` otherwise. The types of ``x`` and ``y`` must be the same.
```

Comment:
Looks like the x and y can be different datatype.
```
spark-sql (default)> select nanvl(cast(10.2 as double), cast(3.4 as float));
10.2
Time taken: 0.13 seconds, Fetched 1 row(s)
```

Reply:
Spark has type coercion rule for this fucntion, so this fucntion should only get either (float, float) or (double, double). We don't need to handle different types inputs.
Thanks.
https://github.com/apache/spark/blob/3a8c1bca4fffaf4185bd0729f24f30e732ba72a4/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala#L497

Reply:
ok